scripts/deploy.py

```diff
@@ -0,0 +1,497 @@
+                                                                 x),
+                fg='red')
+            return None
+    click.secho(
+        'Found current version v%s' % FormatVersion(version), fg='yellow')
+    ctx['old-version'] = version
+    return True
+
+
+def ComputeSha256(filename):
+    BUF_SIZE = 65536
+    sha = hashlib.sha256()
+    with open(filename, 'rb') as f:
+        while True:
+            data = f.read(BUF_SIZE)
+            if not data:
+                return sha.hexdigest()
+            sha.update(data)
+
+
+def RunStoreFileSha(filename):
+    def f(ctx):
+        key = 'self-sha-%s' % filename
+        ctx[key] = ComputeSha256(filename)
+        click.secho("SHA256 of %s is %s" % (filename, ctx[key]))
+        return True
+
+    f.__doc__ = "Store SHA of %s" % filename
+    return f
+
+
+def CompareFileSha(filename, *, require_equal):
+    def f(ctx):
+        sha = ComputeSha256(filename)
+        click.secho("SHA256 of %s is %s" % (filename, sha))
+
+        if require_equal and sha != ctx['self-sha-' + filename]:
+            click.secho("File %s changed." % filename, fg='red')
+            return False
+        if not require_equal and sha == ctx['self-sha-' + filename]:
+            click.secho("File %s not changed." % filename, fg='red')
+            return False
+        return True
+
+    f.__doc__ = "Compare SHA of %s" % filename
+    return f
+
+
+# A step which runs exernal command.
+def RunCmdStep(cmd_line, doc=None):
+    def f(context):
+        click.secho('$ %s' % cmd_line, fg='yellow')
+        r = os.system(cmd_line)
+        if r != 0:
+            click.echo('Return code %d, The command was: %s' %
+                       (r, click.style(cmd_line, fg='yellow')))
+            return False
+        return True
+
+    if doc:
+        f.__doc__ = doc
+    else:
+        f.__doc__ = "Execution of %s " % cmd_line
+    return f
+
+
+def RetryPrompt():
+    while True:
+        click.secho('(type one of: retry/ignore/abort)', fg='cyan')
+        value = click.prompt('', prompt_suffix='>>>>>>> ')
+        if value == 'retry':
+            return True
+        if value == 'ignore':
+            return False
+        if value == 'abort':
+            sys.exit(1)
+
+
+# Exception when step decides to jump to another step.
+class Jump(BaseException):
+    def __init__(self, whereto):
+        self.whereto = whereto
+
+
+class Pipeline:
+    def __init__(self):
+        self.steps = []  # List of steps in a pipeline.
+        self.start = 1  # From which step to start, 1-based.
+        self.end = None  # At which step to end.
+        self.list_only = False  # Only list steps, do not execute.
+        self.step_by_step = False  # Ask confirmation before every step.
+        self.context = {}
+        self.cmd_name = 'unknown'
+
+    def AddStep(self, func):
+        self.steps.append(func)
+
+    def StateFileName(self):
+        return os.path.join('/tmp', '%s_state.json' % self.cmd_name)
```

Comment:
Shouldn't we generate a properly unique tmp file name via the stdlib module for this?

Reply:
Or if not, at least add a lc0 specific portion to the name somewhere

Reply:
It should not be unique, it's (only) intention is to be reused in the case of crash.
Added lc0-specific portion.